### PR TITLE
Explain the Migration strategy and roadmap

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  |Explain the Migration strategy and roadmap in order to clear the confusion about what we need to do about the Domain
| Fixed ticket? | ~

### CODE REVIEW

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+- some business logic classes (Model)
+- ObjectModel classes with lot of logic in it (Model)
+
+We believe PrestaShop project has grown too big to follow this 3-layers model, rather fitted for small applications. We have chosen to introduce more architecture layers.
+
+# The beginning of the Migration project: Configuration Form
+
+The Migration project started by migrating Configuration Form pages.
+
+[You can read more about it by clicking on this link][form-pages].
```
Comment (Progi1984) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:45]:
```suggestion
The Migration project started by migrating [Configuration Form pages][form-pages].
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+The chosen structure to handle such pages was the following:
+
+1. A View layer: Twig template, rendering manually the Symfony form
+2. A controller layer: Symfony Controller empty of business logic (following the ["Thin Controller, Fat Model"][thin-ctrlers] principle), in charge of controlling the View, formatting data, handling security and routing
```
Comment (Progi1984) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:50]:
```suggestion
2. A Controller layer: Symfony Controller empty of business logic (following the ["Thin Controller, Fat Model"][thin-ctrlers] principle), in charge of controlling the View, formatting data, handling security and routing
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+3. In Model layer: Symfony Form is built and managed by a FormHandler
+4. Also part of the Model layer, DataProvider is in charge of data retrieval and saving the data
+
+Since the DataProvider interacts with ObjectModel, it belongs to [Adapter namespace][adapter-namespace].
+
+# Phase 2: listings
+
+For migrating Listing pages, the Grid component was built.
+
+[You can read more about it by clicking on this link][grid-pages].
```
Comment (Progi1984) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:60]:
```suggestion
For migrating Listing pages, the [Grid component][grid-pages] was built.
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+We however wanted to interact with it in a way that would allow us, in the future, to easily unplug it and replace it with a new implementation.
+
+So we decided to introduce [CQRS][cqrs] and [a Domain][domain].
+
+CQRS was introduced into PrestaShop architecture in the following manner:
+
+The Application layer (Forms, Controllers, Views ...) dispatches [Queries and Commands][cqrs] to the Domain to be handled, unaware of the inner workings of this domain.
+
+Inside the domain are QueryHandlers and CommandHandlers that handle dispatched Queries and Commands. For now, they do use PrestaShop legacy business logic to handle it. This means classes like ObjectModel, Cart, Product are still being used inside migrated pages.
```
Comment (Progi1984) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:91]:
```suggestion
Inside the domain are QueryHandlers and CommandHandlers that handle dispatched Queries and Commands. For now, they use PrestaShop legacy business logic to handle it. This means classes like ObjectModel, Cart, Product are still being used inside migrated pages.
```

**Review by Progi1984 — COMMENTED** [review]
Some comments

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
+---
+title: Migration project and strategy
+weight: 10
+---
+
+This is a summary of current Migration Strategy. It provides an overview of our current vision, in the end of this year 2020, of the Symfony Migration project.
```
Comment (atomiix) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:6]:
```suggestion
This is a summary of the current Migration Strategy. It provides an overview of our current vision, in the end of this year 2020, of the Symfony Migration project.
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+# Different kind of pages
+
+Back-Office pages can be classified in four categories:
+
+1. [Configuration / Settings Form pages][form-pages]
+
+These pages allow user to modify configuration settings in PrestaShop.
```
Comment (atomiix) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:14]:
```suggestion
These pages allow the user to modify configuration settings in PrestaShop.
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+2. [Listing pages][grid-pages]
+
+These pages allow user to browse PrestaShop content using listings. These listings usually provide some actions user can trigger, such as "enable/disable", "delete", "bulk delete".
```
Comment (atomiix) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:18]:
```suggestion
These pages allow the user to browse PrestaShop content using listings. These listings usually provide some actions the user can trigger, such as "enable/disable", "delete", "bulk delete".
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+3. [Add/Edit Form pages][identifiable-form-pages]
+
+These pages allow user to create or edit records from PrestaShop data model (example: create/edit customers).
```
Comment (atomiix) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:22]:
```suggestion
These pages allow the user to create or edit records from PrestaShop data model (example: create/edit customers).
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+
+4. Other pages
+
+There are some pages that are unique : Carriers Edit page, Dashboard, Customer Service page ...
```
Comment (atomiix) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:26]:
```suggestion
There are some pages that are unique: Carriers Edit page, Dashboard, Customer Service page ...
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+Sometimes a page will have a mixed content. For example there are pages that provide two listings or one listing and one configuration form.
```
Comment (atomiix) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:28]:
```suggestion
Sometimes a page will have mixed content. For example there are pages that provide two listings or one listing and one configuration form.
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md (hunk 4 above)
Comment (atomiix) [reply src/content/1.7/development/architecture/migration-guide/strategy.md:91]: I think this one has been made on purpose to accentuate that it's using PrestaShop legacy business logic.

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+
+We dont want to render manually Form inputs, so we now aim to build a Twig Form Theme allowing us to render all of our Symfony forms with a single block
+
+```
+{{ form_widget(form) }}
+```
+
+# Towards a Core domain
+
+Between phase 1 and phase 4, two years passed. Two years where we experimented, explored, struggled and we got the hang of what was needed for PrestaShop to mover forward.
```
Comment (atomiix) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:134]:
```suggestion
Between phase 1 and phase 4, two years passed. Two years where we experimented, explored, struggled and we got the hang of what was needed for PrestaShop to move forward.
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+
+{{< figure src="../../img/architecture-overview-future.jpg" title="Future architecture overview" >}}
+
+1. Core Domain
+2. Front-end applications
+3. Contracts and Tools (SDKs)
+4. APIs
+5. Extensions
+
+The **Core** sits in the back-end and is at the center of it all. It accounts for all the business needs and use cases that PrestaShop is capable of doing (managing products, shopping carts, orders, etc.). The Core is _domain-oriented_, meaning that it is built around business use cases, expressed in an [ubiquitous language][ubiquitous-langage]. The Core is also master of its own Domain; in  order to let it be the guardian of system-wide coherence, it has to be isolated from other services and be the only one capable of performing state transitions. Other services can only interact with the Core through well-defined interfaces, that we call the Core API. Incidentally, this also means that Core behavior can be _extended_, not _modified_, by other services–or at least not in a way that it puts system coherence at risk. In addition, the Core is designed to be easily testable and is covered by automated tests.
```
Comment (atomiix) [inline src/content/1.7/development/architecture/migration-guide/strategy.md:148]:
```suggestion
The **Core** sits in the back-end and is at the center of it all. It accounts for all the business needs and use cases that PrestaShop is capable of doing (managing products, shopping carts, orders, etc.). The Core is _domain-oriented_, meaning that it is built around business use cases, expressed in an [ubiquitous language][ubiquitous-langage]. The Core is also master of its own Domain; in  order to let it be the guardian of system-wide coherence, it has to be isolated from other services and be the only one capable of performing state transitions. Other services can only interact with the Core through well-defined interfaces, that we call the Core API. Incidentally, this also means that Core behavior can be _extended_, not _modified_, by other services or at least not in a way that it puts system coherence at risk. In addition, the Core is designed to be easily testable and is covered by automated tests.
```

File: src/content/1.7/development/architecture/migration-guide/strategy.md (hunk 4 above)
Comment (matks) [reply src/content/1.7/development/architecture/migration-guide/strategy.md:91]: Indeed 😉

**Review by atomiix — APPROVED** [review]

File: src/content/1.7/development/architecture/migration-guide/strategy.md (hunk 4 above)
Comment (Progi1984) [reply src/content/1.7/development/architecture/migration-guide/strategy.md:87]: :+1:

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @matks

**Pull request merged**